src/Illuminate/Validation/ValidationServiceProvider.php

```diff
@@ -59,14 +63,30 @@ protected function registerPresenceVerifier()
     }
 
     /**
+     * Register failure formatter.
+     *
+     * @return void
+     */
+    protected function registerFailureFormatter()
+    {
+        $this->app->singleton('validation.failure_formatter', function ($app) {
+            $defaultFormatter = $app['config']['validation']['default_failure_formatter'];
```

Comment:
`$app['config']` can probably be extracted, so that we don't check twice that it is bound in the container - why not:
```php
$config = $app['config'];
$defaultFormatter = $config->get('validation.default_failure_formatter');
$formatter = $config->get('validation.failure_formatters'.$defaultFormatter);
```